services/spar/src/Spar/App.hs

```diff
@@ -172,8 +172,9 @@ bindUser buid userref = do
   unless (uteamid == Just teamid)
     (throwSpar . SparBindFromWrongOrNoTeam . cs . show $ uteamid)
   insertUser userref buid
-  Intra.bindUser buid userref
-  pure buid
+  Intra.bindUser buid userref >>= \case
+    True  -> pure buid
+    False -> throwSpar SparBindUserDisappearedFromBrig
```

Comment:
Can we also log something at least as warning here? This should really never happen and so if it does and we only throw a 404, we're likely to miss this.